src/operation.js

```diff
@@ -24,26 +24,34 @@ export default (context, name, operationName) => async (req, args, options) => {
         operationName,
     });
 
-    return http(
-        buildRequest(
-            requestContext,
-            req,
-            args,
-            options,
-        ),
-    ).then(
-        response => buildResponse(requestContext)(
-            response,
-            requestContext,
-            req,
-            options,
-        ),
-    ).catch(
-        error => buildError(requestContext)(
-            error,
-            requestContext,
-            req,
-            options,
-        ),
+    const request = buildRequest(
+        requestContext,
+        req,
+        args,
+        options,
+    );
+    const retries = get(request, 'retries');
+
+    let errorResponse;
+    for (let attempts = 0; attempts < retries; attempts++) {
```

Comment:
Two points here:
* There may be some requests which are intrinsically non-retryable (certain mutations / POSTS / etc. come to mind). I can limit this to `GET` requests, if it's a concern
* I'm immediately retrying on failure; we can also add a jitter here (e.g. a couple hundred ms) to mitigate.

Reply:
`let attempts = -1`